src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -46,11 +49,18 @@ public void deleteById(Long id) {
 
     public BookDto update(Long id, BookRequestDto bookRequestDto) {
         validateId(id);
-        Book updatedBook = bookMapper.toModel(bookRequestDto);
+        Book updatedBook = bookMapper.toEntity(bookRequestDto);
         updatedBook.setId(id);
         return bookMapper.toDto(bookRepository.save(updatedBook));
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findBooksByCategoryId(Pageable pageable, Long id) {
+        return bookRepository.findAllByCategoryId(pageable, id).stream()
```

Comment:
can we also validate the category here, maybe somehow reuse logic from the `categoryService`?